packages/openapi-ts/src/openApi/shared/transforms/readWrite.ts

```diff
@@ -466,7 +467,8 @@ export const updateRefsInSpec = ({
       let nextPointer = currentPointer;
       let nextContext = context;
       if (isPathRootSchema(path)) {
-        nextPointer = `${schemasPointerNamespace}${path[2]}`;
+        const nameSegment = path[path.length - 1] as string;
+        nextPointer = `${schemasPointerNamespace}${nameSegment}`;
```

Comment:
why is this change needed?

Reply:
This might need some test/comment. I also wonder if this will break with refs like `#/components/schemas/Foo/properties/Bar`